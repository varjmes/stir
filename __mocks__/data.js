export const data = {
  "site": {
    "siteMetadata": {
      "title": "Test title",
      "description": "Test description",
      "author": "Test author"
    }
  }
}

export const indexData = {
  "site": {
    "siteMetadata": {
      "title": "Test title",
      "description": "Test description",
      "author": "Test author"
    }
  },
  "allMarkdownRemark": {
    "totalCount": 2,
    "edges": [
      {
        "node": {
          "id": "123",
          "frontmatter": {
            "title": "Test Post",
            "date": "12 March, 2019",
            "author": "Test author",
            "description": "Test description",
            "keywords": ["test", "test1", "test2"]
          },
          "excerpt": "Test excerpt",
          "fields": {
            "slug": "/test/1/"
          }
        }
      },
      {
        "node": {
          "id": "456",
          "frontmatter": {
            "title": "Test Post",
            "date": "11 March, 2019"
          },
          "excerpt": "Test excerpt",
          "fields": {
            "slug": "/test/2"
          }
        }
      }
    ]
  }
}

export const recipeData = {
  "site": {
    "siteMetadata": {
      "title": "Test Recipe",
      "description": "Test Recipe description",
      "author": "Test recipe author"
    }
  },
  "markdownRemark": {
    "html": "<p>Wow, eggs.</p>",
    "frontmatter": {
      "title": "Eggs"
    }
  }
}
