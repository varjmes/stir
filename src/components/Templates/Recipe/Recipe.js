import React from 'react'
import PropTypes from 'prop-types'
import { graphql } from 'gatsby'

import Layout from '../../Layout/Layout'
import Seo from '../../Seo/Seo'

const Recipe = ({ data }) => {
  const post = data.markdownRemark
  const {title, description, author, date, keywords} = post.frontmatter
  return (
    <Layout>
      <Seo title={title} description={description} author={author} keywords={keywords} />
      <div>
        <h1>{title}</h1>
        <span>{date}</span>
        <div dangerouslySetInnerHTML={{ __html: post.html }} />
      </div>
    </Layout>
  )
}

Recipe.propTypes = {
  data: PropTypes.any.isRequired
}

export default Recipe

export const query = graphql`
  query($slug: String!) {
    markdownRemark(fields: { slug: { eq: $slug } }) {
      html
      frontmatter {
        title
        description
        author
        keywords
        date
      }
    }
  }
`
