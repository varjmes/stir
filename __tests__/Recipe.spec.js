import React from 'react'
import renderer from 'react-test-renderer'
import { useStaticQuery, StaticQuery } from 'gatsby'

import Recipe from '../src/components/Templates/Recipe/Recipe'
import { recipeData } from '../__mocks__/data'


describe('Recipe', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Recipe data={recipeData} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
