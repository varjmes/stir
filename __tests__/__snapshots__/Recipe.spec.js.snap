// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe renders correctly 1`] = `
<main
  className="sc-bdVaJa cHrwRm"
>
  <h1
    className="sc-bwzfXH jotfnM"
  >
    stir.
  </h1>
  <div>
    <h1>
      Eggs
    </h1>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Wow, eggs.</p>",
        }
      }
    />
  </div>
</main>
`;
